json.go

```diff
@@ -121,6 +120,40 @@ func JsonStringIndent(v interface{}, prefix, indent string) string {
 	return string(bs)
 }
 
+// JsonUnmarshal marshals the source into json and unmarshals it into target.
+// If there is an error, it checks for known json parser errors and
+// if there is a match, a JsonUnmarshalError with parsed information is returned.
+// The error messages are formatted by https://pkg.go.dev/encoding/json#Unmarshal
+// * "json: cannot unmarshal <value> into Go value of type <type>"
+// * "json: cannot unmarshal <value> into Go struct field <target property name> of type <type>"
+func JsonUnmarshal(source []byte, target any) (err error) {
+	err = json.Unmarshal(source, target)
+	if err != nil {
+		regex := regexp.MustCompile(`json: cannot unmarshal ([^\s]+) into Go value of type (.+)$`)
+		matches := regex.FindStringSubmatch(err.Error())
+		if len(matches) == 3 {
+			return JsonUnmarshalError().
+				OriginalError(err).
+				SourceType(matches[1]).
+				TargetType(matches[2])
```

Comment:
Das hier ist nicht "errorgen". Schau doch nochmal in meine ursprüngliche Skizze dieses "JsonUnmarshalError", da ist im Prinzip alles was wir brauchen.